checkov/kubernetes/graph_builder/graph_components/edge_builders/ServiceAccountEdgeBuilder.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from checkov.kubernetes.graph_builder.graph_components.blocks import KubernetesBlock
+from checkov.kubernetes.graph_builder.graph_components.edge_builders.K8SEdgeBuilder import K8SEdgeBuilder
+
+
+class VertexConncetions:
+    def __init__(self, origin_vertex_index: int, destination_vertices_indices: list[int] | None = None) -> None:
+        self.origin_vertex_index = origin_vertex_index
+        self.destination_vertices_indices = destination_vertices_indices or []
+
+
+class ServiceAccountEdgeBuilder(K8SEdgeBuilder):
+    def __init__(self) -> None:
+        self._cache = {}
+
+    @staticmethod
+    def should_search_for_edges(vertex: KubernetesBlock) -> bool:
+        return vertex.attributes.get('kind') == 'ServiceAccount'
+
+    def _find_all_service_accounts(self, vertices: list[KubernetesBlock]) -> None:
+        for index, vertex in enumerate(vertices):
+            service_account_name = vertex.attributes.get('metadata.name')
+            if vertex.attributes.get('kind') != 'ServiceAccount' or service_account_name is None:
+                continue
+            self._cache[service_account_name] = VertexConncetions(index)
+
+    def find_connections(self, vertex: KubernetesBlock, vertices: list[KubernetesBlock]) -> list[int]:
+        if self._cache == {}:
+            self._find_all_service_accounts(vertices)
```

Comment:
Your'e right, but I think that as the code in `local_graph` looks now it will make it much more complex.
In `_create_edges` we assume polymorphism, and adding a check just for this type of builder  would break that and I'm not sure the tradeoff is worth it (as now it's also O(n) operation for that resource, rather than O(n^2))

Reply:
On a second look, seems like mypy doesn't allow me to keep it like I did (rightfully so), so I will change it anyway

Reply:
It will also require more computation in the beginning which in most cases will be irrelevant

Reply:
I think I didn't explain it well enough. What I meant, if the call `self._find_all_service_accounts(vertices)` doesn't add a value to the cache, there is no need to try to find vertices with a `destination_vertex_ref`.